wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/Views/Components/DeviceDetailsButtonsView.swift

```diff
@@ -72,14 +72,14 @@ struct DeviceDetailsButtonsView: View {
         if let status = viewModel.e2eIdentityCertificate?.status {
             switch status {
             case .valid:
-                if let isExpiring = viewModel.isCertificateExpiringSoon, isExpiring {
+                if viewModel.isSelfClient, viewModel.isCertificateExpiringSoon == true {
                     Divider()
                     updateCertificateButton.padding()
                 }
                 Divider()
                 showCertificateButton.padding()
             case .notActivated:
-                if !viewModel.isFromConversation {
+                if !viewModel.isFromConversation && viewModel.isSelfClient {
```

Comment:
Instead of repeating `viewModel.isSelfClient` 3 times, you can add it in the first `if`:
`if let status = viewModel.e2eIdentityCertificate?.status, viewModel.isSelfClient {`

Reply:
As we have to show other buttons for all these states, we cannot do this.